feat: user Role 생성

jinseong205 opened the pull request:
### 개요
-구매자 / 판매자 구별에 필요한 role 추가

### 추가 및 변경 사항
- role enum , user entity role 추가
- user 생성시 기본 role 'BUYER' 설정
- user 수정시 role 변경 설정
- user 수정/조회 dto에 role 추가
- 일부 테스트 케이스 수정

resolved #58

jinseong205 committed:
feat: 유저 role 추가

role enum , user erole 추가
role 추가에 따른 일부 service, testcase 수정

jinseong205 committed:
feat: user role 추가 conflict 해결

- conflict 해결

jinseong205 committed:
Merge branch 'develop' of https://github.com/daengdaengLee/until into feat/create-user-role

jinseong205 committed:
feat : User role 추가

import, 중복 된 불필요 library 제거

jinseong205 committed:
feat: 유저 수정 - role 수정 추가

user role 추가에 따른 유저 role 수정 추가

jinseong205 committed:
Merge branch 'develop' of https://github.com/daengdaengLee/until into feat/create-user-role

dudwns9280 commented on src/main/java/hello/until/user/service/UserService.java:
```diff
@@ -19,47 +20,56 @@
 		user.updateEmail(email);
 		user.updatePassword(password);
+
+		if (role != null && !role.equals("")) {
```
이부분은 프론트 단에서 입력한 Role이 저희가 생성한 RoleEnum에 있는지 validation하는 것으로 보여지는데 service단에서 하지 않고, Request단에서 @valid 형식을 통해 유효성을 검증하면 좋을 것 같습니다

jinseong205 replied:
회의 때 나온 내용하고 동일하게 
해당 부분 제거하고 dto에서 role로 받아 예외 처리할 수 있도록 하겠습니다.

dudwns9280 commented on src/main/java/hello/until/user/entity/User.java:
```diff
@@ -26,7 +29,10 @@ public class User {
     private String password;
-
+    
+    @Enumerated(EnumType.STRING)
```
EnumType을 String으로 DB에 저장하시는 이유가 무엇인가요?

jinseong205 replied:
오늘 회의하면서 나온 내용들 간단하게 정리하겠습니다.

EnumType.ORDINAL :  enum 순서 값을 DB에 저장하여 DB 저장공간 축소. 
EnumType.STRING : db 값을 application과 동일하게 가져감으로써 가독성 향상 및 추후 마이그레이션 용이 
 
두 분 다 좋은 의견을 말씀해주셔서 어떤 내용을 고르기가 어려워서 일단 현재 방향으로 가려 합니다!

daengdaengLee commented on src/main/java/hello/until/exception/ExceptionCode.java:
```diff
@@ -12,6 +12,7 @@
 public enum ExceptionCode {
 	NO_USER_TO_GET(HttpStatus.BAD_REQUEST, "해당 유저 정보가 없습니다."),
     NO_USER_TO_UPDATE(HttpStatus.BAD_REQUEST, "수정할 유저가 없습니다."),
+    NO_USER_ROLE_TO_UPDATE(HttpStatus.BAD_REQUEST, "수정할 유저 권한 정보가 잘못되었습니다."),
```
spring validation 으로 처리하고 이 예외 코드는 삭제하는 게 어떨까요?

jinseong205 replied:
넵 이부분도 제거하겠습니다!

daengdaengLee commented on src/main/java/hello/until/user/service/UserService.java:
```diff
@@ -19,47 +20,56 @@
-		User user = this.getUserById(userId).orElseThrow(
-				() -> new CustomException(ExceptionCode.NO_USER_TO_UPDATE)
-		);
+	public User updateUser(long userId, String email, String password, String role) {
```
role 을 String 이 아니라 Role 타입으로 받는 게 좋을 것 같아요.
검증 책임은 컨트롤러에 있을 것 같습니다~

daengdaengLee replied:
그러면 아래 영준님 코멘트 다신 부분도 삭제 가능할 것 같아요.

jinseong205 replied:
회의 때 나온 내용하고 동일하게 
영준님 코멘트 다신 부분 제거하고 dto에서 role로 받아 예외 처리할 수 있도록 하겠습니다.

daengdaengLee commented on src/main/java/hello/until/user/dto/request/UpdateUserRequest.java:
```diff
@@ -1,7 +1,10 @@
         String email,
-        String password){
+        String password,
+        String role){
```
여기를 Role 타입으로 하고 스프링 validation 에서 튕겨내도록 하는 게 좋을 것 같아요.

jinseong205 replied:
넵 해당 내용으로 진행하겠습니다.

daengdaengLee commented on src/main/java/hello/until/user/dto/response/GetUserResponse.java:
UserResponse 와 완전히 동일하게 쓰일 걸로 예상되면 이걸 삭제하고 UserResponse 로 같이 쓰는 것도 좋을 것 같습니다.

jinseong205 replied:
해당 내용들 적용해서 수정하도록 하겠습니다.

daengdaengLee reviewed (commented):
코멘트 확인 부탁드려요~

jinseong205 committed:
Merge branch 'develop' of https://github.com/daengdaengLee/until into feat/create-user-role

jinseong205 committed:
feat: comment 내용 적용

- UpdateUserRequest role String -> Role로 변경
- GetUserRespone 삭제 및 UserResponse로 수정

daengdaengLee reviewed (approved):
LGTM!

이건 이 PR 내용은 아닌데 password 암호화해서 저장하는 거 해야 할 것 같아요. :)

jinseong205 committed:
Merge branch 'develop' into feat/create-user-role
